odl/deform/linearized_deform.py

```diff
@@ -0,0 +1,497 @@
+            Fixed template deformed by the vector field.
+        displacement: `ProductSpaceVector`
+            The point that the adjoint of the Gateaux derivative needs
+            to be computed at.
+
+        Examples
+        --------
+        >>> import odl
+        >>> space = odl.uniform_discr(0, 1, 5)
+        >>> template = space.element([0, 0, 1, 0, 0])
+        >>> disp_field_space = odl.ProductSpace(space, space.ndim)
+        >>> disp_field = disp_field_space.element([[0, 0, 0, -0.2, 0]])
+        >>> vector_field = disp_field_space.element([[1, 1, 1, 2, 1]])
+        >>> op = LinDeforFixedTempDeriv(template, disp_field)
+        >>> op(vector_field)
+        uniform_discr(0.0, 1.0, 5).element([0.0, 5.0, -5.0, -10.0, 0.0])
+        """
+        assert displacement[0] in template.space
+
+        Operator.__init__(self, template.space, displacement.space,
+                          linear=True)
+
+        self.template = template
+        self.displacement = displacement
+
+    def _call(self, func):
+        """Implement ``self(func)```.
+
+        Parameters
+        ----------
+        func : `DiscreteLpVector`
+            The evaluation point, i.e. an element in the template space,
+            for the deformed gradient of the template.
+        """
+        grad = odl.Gradient(self._domain)
+        template_grad = grad(self.template)
+
+        def_grad = deform_grad(template_grad, self.displacement)
+
+        return [gf * func for gf in def_grad]
+
+
+class LinDeforFixedDispOp(Operator):
+    """Operator using fixed displacement to map template to deformed template.
+
+    This linear operator maps a template ``I`` to the corresponding
+    deformed template using a fixed displacement field ``v``.
+    The linearized deformation is given by the left action defined
+    through the pushforward, so
+
+        I --> I(Id + v)
+
+    where the ``I`` is a given template, ``Id`` is the identity mapping:
+
+        y --> y
+
+    and the vector field ``v`` is a displacement mapping:
+
+        y --> v(y)
+
+    Here, the ``y`` is an element in the domain of the target.
+    In diffeomorphic sense, the domain of ``I`` is the same
+    as the target.
+    """
+    def __init__(self, displacement):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        displacement : `ProductSpaceVector`
+            Fixed displacement field used in the linearized deformation.
+
+        Examples
+        --------
+        >>> import odl
+        >>> space = odl.uniform_discr(0, 1, 5)
+        >>> disp_field_space = odl.ProductSpace(space, space.ndim)
+        >>> displacement_field = disp_field_space.element([[0, 0, 0, -0.2, 0]])
+        >>> op = LinDeforFixedDispOp(displacement_field)
+        >>> template = space.element([0, 0, 1, 0, 0])
+        >>> op(template)
+        uniform_discr(0.0, 1.0, 5).element([0.0, 0.0, 1.0, 1.0, 0.0])
+        """
+        assert isinstance(displacement.space, odl.ProductSpace)
```

Comment:
`if not isinstance ... : raise TypeError: ...`

Also need to check `displacement.space[0]` is a `DiscreteLp`